vars/copyAMI.groovy

```diff
@@ -26,13 +31,30 @@ def call(body) {
 
   println "copied ${config.ami} from ${config.region} to ${config.copyRegion} with Id ${copyImageId}"
   copyTags(sourceClient,targetClient,config.ami,copyImageId)
+  if (config.waitOnCopy) {
+    wait(targetClient, config.copyRegion, copyImageId) 
+  }
   return copyImageId
 }
 
 def copyAMI(client,config) {
-  CopyImageRequest copyImageRequest = new CopyImageRequest()
-    .withSourceImageId(config.ami)
-    .withSourceRegion(config.region)
+  CopyImageRequest copyImageRequest
```

Comment:
@timhassett you could make this a bit cleaner by declaring the new CopyImageRequest with the required options and then adding the encrypted flag and kms key id if set.

```java
def copyAMI(client,config) {
  CopyImageRequest copyImageRequest = new CopyImageRequest()
    .withSourceImageId(config.ami)
    .withSourceRegion(config.region)
     
  if (config.encrypted) {
    copyImageRequest.withEncrypted(true)
    if (config.kmsKeyId) {
      copyImageRequest.withKmsKeyId(config.kmsKeyId)
    }
  }

  CopyImageResult copyImageResult = client
    .copyImage(copyImageRequest)

  return copyImageResult.getImageId();
}
```